### PR TITLE
The Bad Guys: Initial work

### PR DESCRIPTION
**Pull request by breville**
Paired work from @breville and @mikeharv...

## Bad Guys Project

In order to partner with Universal to provide an engaging experience for students, new assets were given to us to incorporate into Sprite Lab. As we want to silo these assets from the rest of Sprite Lab, creating a separate **project level** gives us the functionality to do so, along with the benefit of block customizations, and separating this promotion in a clear way for the brand (enabling us to pitch similar partnerships to future brands). This level type will NOT have access to the costumes library (no costumes tab). 

### Engineering Work

At a high level, creating a **new project level** allows us to use Sprite Lab, as is, but with a separate flow for remixing/sharing/publishing. 

This pull request was created after accomplishing the following:

1. Upload new assets to S3
Using the [sprite upload](https://levelbuilder-studio.code.org/sprites) feature on levelbuilder, each new asset was added to the `cdo-animation-library` [bucket](https://s3.console.aws.amazon.com/s3/buckets/cdo-animation-library?region=us-east-1&prefix=level_animations%2F) as level-specific sprites. This enables us to add them as default assets for a project level without adding them to the larger asset library that is available to all Sprite Lab users.
2. Customize JSON for background images
At this time, the Sprite Upload feature doesn't support uploading images for backgrounds. To make an image usable as a background (and not for sprites), we can add the "backgrounds" category to its metadata. The JSON for the single provided background image was manually edited and reuploaded to S3 [here](https://s3.console.aws.amazon.com/s3/object/cdo-animation-library?region=us-east-1&prefix=level_animations/thebadguys_background.json).
3. Create a new project level.
Using a levelbuilder account on the local dashboard server, we cloned `New Sprite Lab Project.level` to create `New The Bad Guys Project.level`. Changes made to this file included:
   - Disabling default animations in favor of the custom animations. This is intended to be supported through the Sprite Upload feature ([Select Start Animations from All Sprites](https://levelbuilder-studio.code.org/sprites/select_start_animations?library=all)), but due to a bug, this level's animation manifest had to be written manually by Mike.
   - Customizing the start blocks, toolbox, instructions, and instructions avatar. If new copy is provided, it can be added to the instructions or in places like `say` blocks. The instructor image is currently one of the character sprite images, but this can be updated easily.
Follow-up item: [STAR-2188: [Bad Guys] Final polish on freeplay level ](https://codedotorg.atlassian.net/browse/STAR-2188)
4. Create a new `make new sprite block`
Because the characters differ drastically by size, it may be challenging for students to create a scene where the size of the sprite is what they'd want or expect. This new block automatically scales each new sprite automatically based on their relative sizes, as indicated in reference images provided by Universal. This block can be added to the start blocks and toolbox of the level.
5. List the new level in `projects_controller.rb`.
This is an incredibly simple change that seems to do the following:
  - Make the level accessible at `projects/thebadguys`
  - Support remixing and sharing of projects created at the above URL without ever bringing users to the standard Sprite Lab project level. All generated links for social media posting were confirmed to use `projects/thebadguys` and not `projects/spritelab`.

## Local Testing Instructions
Because this PR includes changes to various Dashboard files, anyone wish to pull and test locally should run the following commands from their local dashboard directory:
* `bundle exec rake seed:custom_levels LEVEL_NAME="New The Bad Guys Project"`
* `bundle exec rake seed:blocks`
The project level should then be accessible from http://localhost-studio.code.org:3000/projects/thebadguys
Note that if this is not the first time you've loaded the project level, you'll want to reset Version History once there to ensure you are seeing the latest start blocks and animation assets.

## Follow-up Work

* https://github.com/code-dot-org/code-dot-org/pull/45468 
* [STAR-2188: [Bad Guys] Final polish on freeplay level ](https://codedotorg.atlassian.net/browse/STAR-2188)
Mike will be making any final adjustments to the level, including any additional assets or changes to blocks and instructions.
* [STAR-2190: [Bad Guys] Enable marketing changes](https://codedotorg.atlassian.net/browse/STAR-2190)
Kelby will be working to update the homepage, create a landing page, and update the Projects page - Brendan will be assisting and helping ramp up Kelby

### CODE REVIEW

**Review by mikeharv — APPROVED** [review]
Added a detailed description of the work leading up to this PR. These changes will only make the new level accessible via direct URL, so I believe it safe to merge now. Follow-up work that makes changes to the projects gallery or marketing pages should either use a DCDO flag or be merged closer to the 4/11 launch date.

**Review by KylieModen — APPROVED** [review]
LGTM - as we wrap up this work, going back to review the documentation included in this PR and adding any additional learnings so that we could replicate this work in the future will be beneficial.

File: dashboard/config/blocks/GamelabJr/gamelab_makeNewScaledSprite.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+function makeNewScaledSprite(costume,location) {
+  createNewSprite({name: 'temporarySprite'}, costume, location);
+  var size;
+  switch (costume) {
+    case 'wolf':
+    case 'wolf_sunglasses':
+      size = 208 * 8 / 7;
```
Comment (breville) [inline dashboard/config/blocks/GamelabJr/gamelab_makeNewScaledSprite.js:7]: Why `8 / 7`?

Comment (mikeharv) [reply dashboard/config/blocks/GamelabJr/gamelab_makeNewScaledSprite.js:7]: This file has been replaced with new ones found here: 
https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/config/blocks/BadGuys

The calculations there were just done super quickly to get us to some kind of working proof of concept. Each sprite now has a default size value based on its costume and can be automatically scaled while keeping proportions true to what the partner provided.

**Pull request merged**